checker.cpp

```diff
@@ -1,22 +1,51 @@
-#include <assert.h>
-#include <iostream>
-using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7) == true);
-  assert(batteryIsOk(50, 85, 0) == false);
-}
+#include "checker.h"
+#include "checkerTest.h"
+
+bool isBatteryOk(BatterySystem system)
+{
+    TemperatureManager temperatureManager(system.warningLevel, system.language);
+    StateOfChargeManager stateOfChargeManager(system.warningLevel, system.language);
+    ChargeRateManager chargeRateManager(system.warningLevel, system.language);
+    bool temperatureStatus = getBatteryStatus(&temperatureManager, system.temperature, system.displayAlert);
+    bool stateOfChargeStatus = getBatteryStatus(&stateOfChargeManager, system.stateOfCharge, system.displayAlert);
+    bool chargeRateStatus = getBatteryStatus(&chargeRateManager, system.chargeRate, system.displayAlert);
+    return (temperatureStatus && stateOfChargeStatus && chargeRateStatus);
+}
+
+bool getBatteryStatus(BatteryManagementSystem *batteryStatus, float value, void (*displayAlert)(Range, std::vector<std::string>))

```

Comment:
`getBatteryStatus` returns a bool. Hard to make out the meaning of 'true/false'.
Try naming the function so that this becomes apparent.

Reply:
Changed function name to isBatteryStatusWithinLimit.